gauge.mss

```diff
@@ -0,0 +1,562 @@
+
+    [service!=null] {
+      line-width: 1.5;
+    }
+  }
+}
+
+#railway_line_low[zoom<=7]::fill,
+#railway_line_med[zoom=8]::fill,
+#railway_line_fill[zoom>=9]::fill,
+#railway_dual_gauge_line[zoom>=9],
+#railway_multi_gauge_line[zoom>=9] {
+  ["railway"="rail"]["usage"="main"]["service"=null],
+  [zoom>=8]["railway"="rail"]["usage"="branch"]["service"=null],
+  [zoom=10]["railway"="rail"]["usage"="industrial"]["service"=null],
+  [zoom>=13]["railway"="rail"]["usage"=null]["service"=null],
+  [zoom>=11]["railway"="rail"]["usage"="industrial"],
+  [zoom>=11]["railway"="rail"]["usage"=null]["service"="siding"],
+  [zoom>=11]["railway"="rail"]["usage"=null]["service"="crossover"],
+  [zoom>=12]["railway"="rail"]["usage"=null]["service"="yard"],
+  [zoom>=11]["railway"="rail"]["usage"=null]["service"="spur"],
+  [zoom>=10]["railway"="narrow_gauge"]["service"=null],
+  [zoom>=11]["railway"="narrow_gauge"]["service"="spur"],
+  [zoom>=11]["railway"="narrow_gauge"]["service"="siding"],
+  [zoom>=11]["railway"="narrow_gauge"]["service"="crossover"],
+  [zoom>=12]["railway"="narrow_gauge"]["service"="yard"],
+  /* service!=null is required to get a smaller Mapnik XML style with the Carto compiler. */
+  [zoom>=9]["railway"="construction"]["construction_railway"="rail"]["usage"="main"]["service"=null],
+  [zoom>=9]["railway"="construction"]["construction_railway"="rail"]["usage"="branch"]["service"=null],
+  [zoom>=10]["railway"="construction"]["construction_railway"="subway"]["service"=null],
+  [zoom>=10]["railway"="construction"]["construction_railway"="light_rail"]["service"=null],
+  [zoom>=11]["railway"="construction"]["construction_railway"="tram"]["service"=null],
+  [zoom>=13]["railway"="construction"],
+  [zoom>=10]["railway"="subway"]["service"=null],
+  [zoom>=13]["railway"="subway"]["service"!=null],
+  [zoom>=10]["railway"="light_rail"]["service"=null],
+  [zoom>=13]["railway"="light_rail"]["service"!=null],
+  [zoom>=11]["railway"="tram"]["service"=null],
+  [zoom>=13]["railway"="tram"]["service"!=null],
+  [zoom>=10]["railway"="monorail"]["service"=null],
+  [zoom>=13]["railway"="monorail"]["service"!=null],
+  [zoom>=11]["railway"="miniature"]["service"=null],
+  [zoom>=13]["railway"="miniature"]["service"!=null] {
+
+    /* fill rules for monorail and miniature */
+    [zoom>=10]["railway"="monorail"]["service"=null],
+    [zoom>=13]["railway"="monorail"]["service"!=null],
+    [zoom>=11]["railway"="miniature"]["service"=null],
+    [zoom>=13]["railway"="miniature"]["service"!=null] {
+      line-width: 3;
+
+      [service!=null] {
+        line-width: 1.5;
+      }
+    }
+
+    /* default color for unknown gauge value */
+    line-color: @color_gauge_unknown;
+
+    #railway_dual_gauge_line {
+      line-dasharray: @dual-gauge-dashes;
+    }
+    
+    #railway_multi_gauge_line {
+      line-dasharray: @multi-gauge-dashes;
+    }
+    
+    /* tracks with numeric gauge value */
+
+    [gauge=380],
+    [gauge>380][gauge<500] {
```

Comment:
I make use of the railway_to_int function, and replaced the matching of gauge value by the integer field.
Originally, I started to develop and test the mapcss via overpass-turbo, but it doesn't work well with the >= operator. 
I have chosen the limit values for the colors in such a way that the gauge in most cases matches the limit values exactly. If e.g. the gauge is equal to 1000, then a rule like [gauge=1000] would apply immediately, but
[gauge>1000] [gauge<1009] is never used so well because the values between the limits are very rare or do not exist in the world.